modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -229,7 +244,7 @@ inline void preprocess(const cv::Mat& src,
             pp = rsz;
         }
 
-        if (!is_hwc && new_c > 1) {
+        if (!is_hwc && new_c >= 1) {
```

Comment:
Isn't it better when grayscale `pp` is just explicitly moved to `dst` in `else`? in `if` it is also moved there, but, I mean, non-trivially

Reply:
?

Reply:
I don't see why you should add `>=` here: if input is 1-channeled, there is no planes it should be divided to, so it should go to `else` branch and just be copied to dst

Reply:
Fixed. I understood.